modules/system/classes/PluginBase.php

```diff
@@ -256,17 +256,13 @@ public function registerMailPartials()
      * Registers a new console (artisan) command
      *
      * @param string $key The command name
-     * @param string $class The command class
+     * @param string|\Closure $command The command class or closure
      * @return void
      */
-    public function registerConsoleCommand($key, $class)
+    public function registerConsoleCommand($key, $command)
     {
         $key = 'command.'.$key;
-
-        $this->app->singleton($key, function ($app) use ($class) {
-            return new $class;
-        });
-
+        $this->app->singleton($key, $command);
```

Comment:
@martendej Would you mind changing this to the following, it will be more backwards-compatible:
```suggestion
        if (is_string($command)) {
            $closure = function ($app) use ($command) {
                return new $command;
            };
        } else {
            $closure = $command;
        }

        $this->app->singleton($key, $closure);
```

Reply:
@martendej Ah never mind, I see where this all goes. Please disregard my requested change.